src/scripts/main.js

```diff
@@ -1,3 +1,318 @@
 'use strict';
 
-// write your code here
+const startButton = document.querySelector('button.start');
+const controls = document.querySelector('.controls');
+const score = document.querySelector('.game-score');
+
+const messangeStart = document.querySelector('p.message.message-start');
+const messangeWin = document.querySelector('p.message.message-win');
+const messangeLose = document.querySelector('p.message.message-lose');
+
+const cells = document.getElementsByClassName('field-cell');
+const rows = document.getElementsByTagName('tr');
+const arrayRows = [...rows];
+const columnsArray = createColumns();
+
+function appearCell() {
+  const myArray = [...cells];
+
+  const freeCells = myArray.filter(cell => cell.textContent === '');
+
+  const randomItem = freeCells[Math.floor(Math.random() * freeCells.length)];
+  const x = Math.random(1 - 4);
+
+  if (freeCells.length === 0) {
+    return;
+  }
+
+  if (x < 0.1) {
+    randomItem.textContent = '4';
+    randomItem.classList.add('field-cell--4');
+  } else {
+    randomItem.textContent = '2';
+    randomItem.classList.add('field-cell--2');
+  }
+}
+
+function scoreCounter(sum) {
+  const previousScore = +score.textContent;
+  const newScore = +sum + previousScore;
+
+  score.textContent = newScore;
+};
+
+startButton.addEventListener('click', () => {
+  appearCell();
+  appearCell();
+
+  const restartButton = document.createElement('button');
+
+  restartButton.classList.add('button');
+  restartButton.classList.add('restart');
+  restartButton.id = 'restart';
+  restartButton.textContent = 'restart';
+  controls.append(restartButton);
+
+  startButton.remove();
+
+  restartButton.addEventListener('click', () => {
+    for (const cell of cells) {
+      cell.textContent = '';
+      cell.className = '';
+      cell.classList.add('field-cell');
+    };
+    appearCell();
+    appearCell();
+  });
+
+  messangeStart.classList.add('hidden');
+});
+
+document.addEventListener('keydown', function(ev) {
+  const key = ev.key;
+  const button = document.querySelector('button');
+
+  if (key === 'ArrowRight') {
+    moveRight();
+    toRight();
+    moveRight();
+    moveRight();
+  }
+
+  if (key === 'ArrowLeft') {
+    moveLeft();
+    toLeft();
+    moveLeft();
+    moveLeft();
+  }
+
+  if (key === 'ArrowUp') {
+    moveUp();
+    toUp();
+    moveUp();
+    moveUp();
+  }
+
+  if (key === 'ArrowDown') {
+    moveDown();
+    toDown();
+    moveDown();
+    moveDown();
+  }
+
+  if (button.className.includes('restart')) {
+    appearCell();
+  };
+
+  winMessange();
+  loseMessange();
+});
+
+function toRight() {
+  arrayRows.forEach((el) => {
+    for (let i = 3; i >= 0; i--) {
+      const rowCells = el.children;
+
+      const theCell = rowCells[i];
+      const prevCell = rowCells[i - 1];
```

Comment:
```suggestion
      const [prevCell, currentCell] = [rowCells[i - 1] , rowCells[i]];
```
bit shorter
can do this everywhere